cwms-data-api/src/main/java/cwms/cda/api/StreamController.java

```diff
@@ -0,0 +1,292 @@
+import static cwms.cda.api.Controllers.queryParamAsClass;
+import static cwms.cda.api.Controllers.requiredParam;
+import cwms.cda.api.enums.UnitSystem;
+import cwms.cda.data.dao.DeleteRule;
+import cwms.cda.data.dao.JooqDao;
+import cwms.cda.data.dao.StreamDao;
+import cwms.cda.data.dto.stream.Stream;
+import cwms.cda.formatters.ContentType;
+import cwms.cda.formatters.Formats;
+import io.javalin.apibuilder.CrudHandler;
+import io.javalin.core.util.Header;
+import io.javalin.http.Context;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiParam;
+import io.javalin.plugin.openapi.annotations.OpenApiRequestBody;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import org.jetbrains.annotations.NotNull;
+import org.jooq.DSLContext;
+
+import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+
+import static cwms.cda.data.dao.JooqDao.getDslContext;
+
+public final class StreamController implements CrudHandler {
+
+    static final String TAG = "Streams";
+
+    private final MetricRegistry metrics;
+    private final Histogram requestResultSize;
+
+    public StreamController(MetricRegistry metrics) {
+        this.metrics = metrics;
+        String className = this.getClass().getName();
+        requestResultSize = this.metrics.histogram((name(className, RESULTS, SIZE)));
+    }
+
+    private Timer.Context markAndTime(String subject) {
+        return Controllers.markAndTime(metrics, getClass().getName(), subject);
+    }
+
+    @OpenApi(
+            queryParams = {
+                    @OpenApiParam(name = OFFICE, required = true, description = "Office id for the reservoir project location " +
+                            "associated with the streams."),
+                    @OpenApiParam(name = DIVERTS_FROM_STREAM_ID, description = "Specifies the stream-id of the " +
+                            "stream that the returned streams flow from."),
+                    @OpenApiParam(name = FLOWS_INTO_STREAM_ID, description = "Specifies the stream-id of the " +
+                            "stream that the returned streams flow into."),
+                    @OpenApiParam(name = UNIT_SYSTEM, type = UnitSystem.class, description = UnitSystem.DESCRIPTION)
+            },
+            responses = {
+                    @OpenApiResponse(status = STATUS_200, content = {
+                            @OpenApiContent(isArray = true, type = Formats.JSONV1, from = Stream.class),
+                            @OpenApiContent(isArray = true, type = Formats.JSON, from = Stream.class)
+                    })
+            },
+            description = "Returns matching CWMS Stream Data for a Reservoir Project.",
+            tags = {TAG}
+    )
+    @Override
+    public void getAll(Context ctx) {
+        String office = ctx.queryParam(OFFICE);
```

Comment:
office is marked required by the OpenAPI docs, but isn't getting validated

Reply:
This should be an office mask. Updated